redaxo/src/core/lib/console/assets_sync.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @package redaxo\core
+ * 
+ * @author staabm
+ *
+ * @internal
+ */
+class rex_command_assets_sync extends rex_console_command
+{
+    protected function configure()
+    {
+        $this
+            ->setDescription('Sync folders and files of /assets with /redaxo/src/addons/my-addon/assets (or plugin) respectively /redaxo/src/core/assets folders');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $created = $updated = $errored = 0;
+        $io = $this->getStyle($input, $output);
+
+        foreach(rex_package::getInstalledPackages() as $package) {
+            $assetsPublicPath = $package->getAssetsPath();
+            $assetsSrcPath = $package->getPath('assets/');
+
+            if (!is_dir($assetsPublicPath) && !is_dir($assetsSrcPath)) {
+                continue;
+            }
+            if (!is_dir($assetsPublicPath)) {
+                rex_dir::create($assetsPublicPath);
+            }
+            if (!is_dir($assetsSrcPath)) {
+                rex_dir::create($assetsSrcPath);
+            }
+
+            // sync 1st way, copies ...
+            // - existing in FE but not "src"
+            // - newer in FE then "src"
+            // - newer in "src" then FE
+            list($ctd, $upd, $err) = $this->sync($io, $assetsPublicPath, $assetsSrcPath);
+            $created += $ctd;
+            $updated += $upd;
+            $errored += $err;
+
+            // sync 2nd way, copies ...
+            // - existing in "src" but not FE
+            list($ctd, $upd, $err) = $this->sync($io, $assetsSrcPath, $assetsPublicPath);
+            $created += $ctd;
+            $updated += $upd;
+            $errored += $err;
+        }
+
+        list($ctd, $upd, $err) = $this->sync($io, rex_path::coreAssets(), rex_path::core('assets/'));
+        $created += $ctd;
+        $updated += $upd;
+        $errored += $err;
+
+        if (!$io->isVerbose()) {
+            $io->success(sprintf('Successfully created %s, updated %s file(s) while running into %s errors.', $created, $updated, $errored));

```

Comment:
Die Meldung würde ich ruhig immer ausgeben, so als Gesamt-Ergebnis ist es auch in der Verbose-Ausgabe am Ende sinvoll, finde ich.